### PR TITLE
Add constants for precompiles addresses range.

### PR DESCRIPTION
**Pull request by lprovenzano**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

- I have changed the magic numbers to constants.
- Refactor in `is_precompile()` for better cohesion of comparisons.
- Added unit tests to make sure it behaves as it should.

## What is the new behavior?

It does not change the behavior, it adds better readability.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot-ssj/868)

### CODE REVIEW

File: crates/evm/src/model.cairo [hunk 1]
```diff
@@ -139,7 +143,10 @@ impl AddressImpl of AddressTrait {
     /// Check whether an address for a call-family opcode is a precompile.
     fn is_precompile(self: EthAddress) -> bool {
         let self: felt252 = self.into();
-        return (self != 0 && (self.into() < 10_u256 || self.into() == 256_u256));
+        let not_equal_than_zero: bool = self != ZERO;
+        let less_than_limit: bool = self.into() < LIMIT_PRECOMPILE_ADDRESS;
+        let equal_than_max: bool = self.into() == MAX_PRECOMPILE_ADDRESS;
```
Comment (enitrat) [inline crates/evm/src/model.cairo:148]: no need for extra variables, just inline the comparisons

File: crates/evm/src/model.cairo [hunk 2]
```diff
@@ -13,6 +13,10 @@ use utils::helpers::{ResultExTrait};
 use utils::set::{Set, SpanSet};
 use utils::traits::{EthAddressDefault, ContractAddressDefault, SpanDefault};
 
+const MAX_PRECOMPILE_ADDRESS: u256 = 256;
```
Comment (enitrat) [inline crates/evm/src/model.cairo:16]: prefer hexadecimal
also: a better name would be the one from the reference.
```suggestion
const FIRST_ROLLUP_PRECOMPILE_ADDRESS: u256 = 0x100;
```

File: crates/evm/src/model.cairo [hunk 3]
```diff
@@ -13,6 +13,10 @@ use utils::helpers::{ResultExTrait};
...
+const LIMIT_PRECOMPILE_ADDRESS: u256 = 10;
```
Comment (enitrat) [inline crates/evm/src/model.cairo:17]:
```suggestion
const LIMIT_PRECOMPILE_ADDRESS: u256 = 0x0a;
```

File: crates/evm/src/model.cairo [hunk 4]
```diff
@@ -404,8 +410,55 @@ mod tests {
...
 
-        // Then
         // Then
         assert(balance == native_token.balanceOf(ca_address.starknet), 'wrong balance');
     }
+
+    #[test]
+    fn test_is_precompile(){
+       // Given
+       let evm_address: EthAddress = 5.try_into().unwrap();
```
Comment (enitrat) [inline crates/evm/src/model.cairo:420]: you could create a single test with all the valid precompiles in an array and ensure that, for each address, it's valid.

```suggestion
   let valid_precompiles = [0x1,0x2,0x3,0x4,0x5,0x6,0x7,0x8,0x9,0xa,0x100];
   for address in valid_precompiles {
      assert!((*address).try_into().unwrap().is_precompile());
     }
  ```
  
  (syntax to verify)
  
  This avoid creating a lot of tests that don't have extensive coverage and take some time to setup and run
```

File: crates/evm/src/model.cairo [hunk 5]
```diff
@@ -404,8 +410,55 @@ mod tests {
...
+
+       // When
+       let is_precompile = evm_address.is_precompile();
+
+       // Then
+       assert_eq!(true, is_precompile, "expected: {:?}, got: {:?}", true, is_precompile);
```
Comment (enitrat) [inline crates/evm/src/model.cairo:426]: just for your knowledge, you don't have to add the "expected ... got ..." string, assert_eq! handles this automatically

**Review by enitrat — CHANGES_REQUESTED** [review]

**Comment by lprovenzano** [comment]
Done changes @enitrat.

File: crates/evm/src/model.cairo [hunk 6]
```diff
@@ -415,13 +413,20 @@ mod tests {
...
+       let mut i = 0;
+       let valid_precompiles = array![0x1,0x2,0x3,0x4,0x5,0x6,0x7,0x8,0x9,0x100];
+       let to = valid_precompiles.len() - 1;
 
        // When
-       let is_precompile = evm_address.is_precompile();
-
-       // Then
-       assert_eq!(true, is_precompile, "expected: {:?}, got: {:?}", true, is_precompile);
+       loop {
```
Comment (enitrat) [inline crates/evm/src/model.cairo:420]: use a `for` loop instead https://book.cairo-lang.org/ch02-05-control-flow.html?highlight=for%20loop#looping-through-a-collection-with-for

Comment (lprovenzano) [reply crates/evm/src/model.cairo]: Done.

**Review by enitrat — APPROVED** [review]

**Pull request merged**